### PR TITLE
Refactor, geocoding and bearing

### PR DESCRIPTION
**Pull request by chris-j-major**
Refactors out some code from index.js, specificaly logging.js and geo.js

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,87 +22,76 @@ var username = process.env.PGO_USERNAME;
...
+                        request.post({
+                          url: process.env.SLACK_WEBHOOK_URL,
+                          json: true,
+                          body: {
+                            text: message,
+                            icon_url: pokemon.img
+                          }
+                        }, function(error, response, body) {
+                          if(error) logger.error(error);
+                          if(response.body) console.log(response.body);
```
Comment (kmoe) [inline index.js]: `console.log` -> `logger.log`

**Comment by kmoe** [comment]
@chris-j-major merge conflicts with master appeared after merging your other PR - let me know if you have any questions

File: geo.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  var s = toRadians(start);
+  var d = toRadians(target);
+  var dLong  = d.longitiude - s.longitiude;
+  var inner = 
+      Math.tan(d.latitude/2.0+Math.PI/4.0)/
+      Math.tan(s.latitude/2.0+Math.PI/4.0);
+  var dPhi = Math.log(inner);
+  if (Math.abs(dLong) > Math.PI ){
+    if (dLong > 0.0)
+      dLong = -2(2*Math.PI - dLong);
```
Comment (kmoe) [inline geo.js]: surely this throws?

File: index.js [hunk 3]
```diff
@@ -37,86 +21,112 @@ var username = process.env.PGO_USERNAME;
...
-              pokeMap = newPokeMap;
             }
           }
-        }
+          logger.log('info','Found '+hbPokemon.length+' pokemon');
+          if ( hbPokemon.length == 0 ) return;
+          var newPokemon = removeKnownPokemon( hbPokemon );
+          logger.log('info','Found '+newPokemon.length+' new pokemon');
+          if ( newPokemon.length == 0 ) return;
+          var interestingPokemon = removeUniteretingPokemon( newPokemon );
```
Comment (kmoe) [inline index.js]: typo in `removeUniteretingPokemon`

File: index.js [hunk 4]
```diff
@@ -37,86 +21,112 @@ var username = process.env.PGO_USERNAME;
...
       });
-    }, 60000);
+    }
+    getHeartbeat();
+    setInterval( getHeartbeat , 60000);
   });
 });
+
+
+var knownPokemon = {};
```
Comment (kmoe) [inline index.js:88]: This is basically an imperative version of the `pokeMap`\- any reason not to implement this logic with lodash?

Comment (chris-j-major) [inline index.js:88]: Yes it is - I'm not too familiar with lodash, and wanted to write something I knew would work.

**Pull request merged**